Applications/spire/source/time_and_sales/time_and_sales_window.cpp

```diff
@@ -55,6 +257,90 @@ void time_and_sales_window::closeEvent(QCloseEvent* event) {
+  if(!m_empty_window_label->isHidden()) {
+    widget_geometry = QRect(
+      m_empty_window_label->mapToGlobal(
+        m_empty_window_label->geometry().topLeft()),
+      m_empty_window_label->mapToGlobal(
+        m_empty_window_label->geometry().bottomRight()));
+  } else {
+    widget_geometry = QRect(
+      m_table->mapToGlobal(m_table->geometry().topLeft()),
+      m_table->mapToGlobal(m_table->geometry().bottomRight()));
+  }
+  if(widget_geometry.contains(event->globalPos())) {
+    m_context_menu->exec(event->globalPos());
+  }
+}
+
+bool time_and_sales_window::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == m_table ||
+      watched == m_table->horizontalHeader()->viewport()) {
+    if(event->type() == QEvent::Wheel) {
+      auto e = static_cast<QWheelEvent*>(event);
+      if(e->modifiers() & Qt::ShiftModifier) {
+        if(m_table->verticalScrollBar()->isVisible()) {
+          m_table->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+        }
+        m_table->setHorizontalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+        m_table->horizontalScrollBar()->setValue(
+            m_table->horizontalScrollBar()->value() - e->delta() / 2);
+        m_h_scrolling = true;
+        m_h_scroll_bar_timer->start();
+      } else {
+        if(!within_h_scroll_bar(e->pos())) {
+          if(m_table->horizontalScrollBar()->isVisible()) {
+            m_table->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+          }
+          m_table->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+          m_table->verticalScrollBar()->setValue(
+            m_table->verticalScrollBar()->value() - e->delta() / 2);
+          m_v_scrolling = true;
+          m_v_scroll_bar_timer->start();
+        }
+      }
+    } else if(event->type() == QEvent::MouseMove) {
+      auto e = static_cast<QMouseEvent*>(event);
+      if(e->pos().x() > width() - m_table->verticalScrollBar()->width()) {
+        m_table->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+        m_table->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+      } else if(within_h_scroll_bar(e->pos())) {
+        m_table->setHorizontalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+        m_table->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+      } else {
+        if(!m_v_scrolling &&
+            m_table->verticalScrollBarPolicy() != Qt::ScrollBarAlwaysOff) {
+          fade_out_vertical_scroll_bar();
+        }
+        if(!m_h_scrolling &&
+            m_table->horizontalScrollBarPolicy() != Qt::ScrollBarAlwaysOff) {
+          fade_out_horizontal_scroll_bar();
+        }
+      }
+    }
+  } else if(watched == m_table->verticalScrollBar()) {
+    if(event->type() == QEvent::HoverLeave) {
+      fade_out_vertical_scroll_bar();
+    }
+  } else if(watched == m_table->horizontalScrollBar()) {
+    if(event->type() == QEvent::HoverLeave) {
+      fade_out_horizontal_scroll_bar();
+    }
+  } else if(watched == QWidget::window()) {
```

Comment:
As previously noted, see if this can be placed on the table rather than the window.